DataBiosphere/terra-workspace-manager
# [PF-1681] Add discoverer policy when creating SAM workspace resource

This PR:
- Changes WSM to understand SAM discoverer role
- When a workspace is created, adds empty discoverer policy. (Since policies can only be added when a SAM resource is created.)

Future PRs will modify ListWorkspaces/GetWorkspace to add a request permission for returning discoverer workspaces. For now, ListWorkspaces *does not* return a workspace if caller is only discoverer.

Manually tested:

- Created workspace. Can see empty discoverer policy with getRoles.
- Called grantRole to add second user as discoverer
- Second user calls GetWorkspace. GetWorkspace returns 403 as expected
- At first, when second user called ListWorkspaces, discoverer workspace was returned, which was wrong. I made the change to SamService.java. Now when second user calls ListWorkspace, discoverer workspace is not returned.

WorkspaceApiControllerTest.java uses mock bio.terra.workspace.service.iam.SamService. I wanted my tests to use real bio.terra.workspace.service.iam.SamService, so I created WorkspaceApiControllerConnectedTest.java.
